### PR TITLE
feat: add stats detailed option

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
added stats detailed option which was unrecognized.

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
Yes

**Summary**
- Added detailed stats option which we missed.

**Does this PR introduce a breaking change?**
No

**Other information**
NA

### CODE REVIEW

**Comment by anshumanv** [comment]
@rishabh3112 here - https://github.com/webpack/webpack/blob/f0b1a71d5a4ba7a685e76d08281195364d7dd9b3/lib/stats/DefaultStatsPresetPlugin.js#L47

Probably it needs to be documented too, I'll open an issue in the docs?

File: packages/webpack-cli/lib/groups/StatsGroup.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const { logger } = require('@webpack-cli/logger');
  */
 class StatsGroup extends GroupHelper {
     static validOptions() {
-        return ['minimal', 'none', 'normal', 'verbose', 'errors-warnings', 'errors-only'];
+        return ['minimal', 'none', 'normal', 'verbose', 'errors-warnings', 'errors-only', 'detailed'];
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/groups/StatsGroup.js]: @rishabh3112 All values are valid https://github.com/webpack/webpack/blob/master/schemas/WebpackOptions.json#L2605

Comment (evilebottnawi) [inline packages/webpack-cli/lib/groups/StatsGroup.js]: @anshumanv Can you reorder values as here https://github.com/webpack/webpack/blob/master/schemas/WebpackOptions.json#L2605?

**Comment by rishabh3112** [comment]
Thanks for in site @anshumanv. It is good now :)

File: packages/webpack-cli/lib/groups/StatsGroup.js (hunk 1 above)
Comment (rishabh3112) [reply packages/webpack-cli/lib/groups/StatsGroup.js]: Why not extract it from there ?

Comment (evilebottnawi) [reply packages/webpack-cli/lib/groups/StatsGroup.js]: @rishabh3112 We will do it late, it is part of task - `Args from webpack`

**Comment by anshumanv** [comment]
@evilebottnawi @rishabh3112, should be good now. :)

File: packages/webpack-cli/lib/groups/StatsGroup.js (hunk 1 above)
Comment (rishabh3112) [reply packages/webpack-cli/lib/groups/StatsGroup.js]: Okay!

**Pull request merged**